vaas/vaas/cluster/models.py

```diff
@@ -21,26 +21,44 @@ class LogicalCluster(models.Model):
     error_timestamp = models.DateTimeField(default=timezone.now)
     last_error_info = models.CharField(max_length=400, null=True, blank=True)
     current_vcl_versions = models.CharField(max_length=400, default='[]')
+    labels_list = models.CharField(max_length=500, default='[]')
     partial_reload = models.BooleanField(default=False)
     service_mesh_routing = models.BooleanField(default=False)
     _current_vcls = None
+    _labels = None
 
     @property
     def current_vcls(self):
         if self._current_vcls is None:
-            try:
-                self._current_vcls = set(json.loads(self.current_vcl_versions))
-            except:  # noqa
-                self._current_vcls = set()
+            self._current_vcls = self._get_set(self.current_vcl_versions)
         return self._current_vcls
 
     @current_vcls.setter
     def current_vcls(self, versions):
-        if isinstance(versions, set):
-            versions = list(versions)
-        if isinstance(versions, list):
-            self._current_vcls = set(versions)
-            self.current_vcl_versions = json.dumps(versions)
+        self._current_vcls, self.current_vcl_versions = self._prepare_set_and_json(versions)
+
+    @property
+    def labels(self):
+        if self._labels is None:
+            self._labels = self._get_set(self.labels_list)
+        return self._labels
+
+    @labels.setter
+    def labels(self, labels):
+        self._labels, self.labels_list = self._prepare_set_and_json(labels)
+
+    def _get_set(self, value):
```

Comment:
A little ambiguous name, don't you think?

Reply:
Yes. I have had same feeling. I don't know what name will fit here. Do you have some proposition?

Reply:
Ok. Never mind. I refactor in different way :)